vendor/github.com/grafana/e2e/util.go

```diff
@@ -256,3 +258,200 @@ func GetTempDirectory() (string, error) {
 
 	return absDir, nil
 }
+
+// based on GenerateTestHistograms in github.com/prometheus/prometheus/tsdb
+func GenerateTestHistogram(i int) *histogram.Histogram {
```

Comment:
This looks a copy of `tsdb.GenerateTestHistograms()`. Any reason why we just don't use it? If it's because you didn't want to add the prometheus dependency to grafana/e2e (which I agree), I don't even think these utilities should be in grafana/e2e. Since they look specific to Mimir e2e testing, why not having them directly in this repo at `integration/`?

_Same comment applies below._

Reply:
It's not a direct copy - well yet, I'm trying to get https://github.com/prometheus/prometheus/pull/11911 merged.
We already have prometheus/prometheus as dependency of grafana/e2e - way before the histograms work.

The idea to have it here is to be consistent with the existing functions.
And we copied it because we also have the promql result version of them , which prometheus doesn't have so the generator and expected result are ensured to be the same.

I can move these to mimir/integration and reuse prometheus code if you think that's better - that would reduce the impact if prometheus changes the generator function and we need to fix the expected result - in fact I'd add a unit test on it.

Reply:
My main comment is that I don't think generating Mimir-specific features belong to grafana/e2e. You mentioned it's not a direct copy, but isn't just about calling `tsdb.GenerateTestHistograms(1)` and then get the 1st result to get the same exact behaviour?